src/ProtobufKdb.cpp

```diff
@@ -160,6 +217,75 @@ K ListImportedMessageTypes(K unused)
   return message_list;
 }
 
+K SetStringKdbType(K kdb_type)
+{
+  if (kdb_type->t != -KH)
+    return krr((S)"kdb_type not -KH");
+
+  switch (kdb_type->h) {
+  case -KS:
+  case KG:
+  case KC:
+    kx::protobufkdb::KdbTypes::Instance()->SetStringKdbType(kdb_type->h);
+    break;
+  default:
+    return krr((S)"kdb_type value not -11|4|10");
+  }
+
+  return (K)0;
+}
+
+K GetStringKdbType(K unused)
+{
+  return kh(kx::protobufkdb::KdbTypes::Instance()->GetStringKdbType());
+}
+
+K SetStringMapKeyKdbType(K kdb_type)
+{
+  if (kdb_type->t != -KH)
+    return krr((S)"kdb_type not -KH");
+
+  switch (kdb_type->h) {
+  case -KS:
+  case KG:
+  case KC:
+    kx::protobufkdb::KdbTypes::Instance()->SetStringMapKeyKdbType(kdb_type->h);
+    break;
+  default:
+    return krr((S)"kdb_type value not -11|4|10");
+  }
+
+  return (K)0;
+}
+
+K GetStringMapKeyKdbType(K unused)
+{
+  return kh(kx::protobufkdb::KdbTypes::Instance()->GetStringMapKeyKdbType());
+}
+
+K SetBytesKdbType(K kdb_type)
+{
+  if (kdb_type->t != -KH)
+    return krr((S)"kdb_type not -KH");
+
+  switch (kdb_type->h) {
+  case -KS:
+  case KG:
+  case KC:
+    kx::protobufkdb::KdbTypes::Instance()->SetBytesKdbType(kdb_type->h);
+    break;
+  default:
+    return krr((S)"kdb_type value not -11|4|10");
```

Comment:
How about
```
"value of kdb_type must be either of -11|4|10h"
```

Reply:
Is there any situation where user want to cast string to kdb bytes? Looks like bytes are expected to specify bytes as its field.
```
q).protobufkdb.SetStringKdbType[4h]
q)scalars:(12i;55f;0x1234ab);
q)serialized:.protobufkdb.serializeArrayFromList[`ScalarExample; scalars];
[libprotobuf ERROR google/protobuf/wire_format_lite.cc:577] String field 'ScalarExample.scalar_string' contains invalid UTF-8 data when serializing a protocol buffer. Use the 'bytes' type if you intend to send raw bytes. 
```

Reply:
That's a limitation of protobuf.  String fields have to be utf8 encoded.  Bytes fields can be anything.